docs/new/develop/indexes.md

```diff
@@ -0,0 +1,99 @@
+
+Consider that you want to retrieve all the Ethereum events matching a specific address. Usually, in every block, you would iterate through all the logs  looking for those where `log.address == ADDRESS`.
+
+With indexing, you could have a pre-cached module with the information of all the event addresses in the block. Instead of reading the full Ethereum block, you can search in the events index (event cache) and avoid decoding the data of those blocks that do not contain events you are interested in.
+
+In the following diagram, you can see three blocks with their corresponding data. Consider that you want to retrieve all the events where `log.address == 0xcd2...`. In a non-indexed module, you would have to decode the data of every block, iterate over the logs, and search if there is any log matching the condition.
+
+<figure><img src="../../.gitbook/assets/develop/cache-block-1.png" width="100%" /></figure>
+
+On the other hand, in an indexed module, the events of every block are pre-cached in a special store, so when you look for events where `log.address == 0xcd2...`, you can simply search in the index store of the block. If the event is contained, within the block, then you decode the data. If not, you skip it.
+
+In the following diagram, `Block 1` and `Block 2` contain an event where `log.address == 0xcd2...`, but `Block 3` does not.
+
+<figure><img src="../../.gitbook/assets/develop/cache-block-2.png" width="100%" /></figure>
+
+### Create a Custom Index
+
+Anyone can create an indexed module. All you need to do is create a Substreams that indexes the piece of data of your choice. For example, let's take a look at the `index_events` module from the [Ethereum Foundational Modules GitHub repository](https://github.com/streamingfast/substreams-foundational-modules/blob/develop/ethereum-common/substreams.yaml#L53).
+
+The flow of indexing the events of a block looks like the following:
+
+<figure><img src="../../.gitbook/assets/develop/cache-flow.png" width="100%" /></figure>
+
+1. The `all_events` module receives a `Block` object as an input and outputs an `Events`, with all the events of the block.
+2. The `index_events` module receives the `Events` object as an input and outputs a `Keys` object, containing the `address` and `signature` fields of every event you want to track.
+3. The `filtered_events` receives the `index_events` module as an input plus a string with the event addresses that the Substreams must filter.
+Given this string of addresses, Substreams checks if the event address is contained on a given block before actually decoding the data of the block.
+You can use logical operators (`and` and `or`) to select what events to search.
+
+The definition of the `index_events` module looks like any other Substreams module, but it is a special _kind_, `kind: blockIndex` and outputs a special data model, `sf.substreams.index.v1.Keys`. The `Keys` object contains the data that must be cached.
```

Comment:
> The `Keys` object contains the data that must be cached.
would be clearer to explain as:
-> contains a list of labels that will be used to identify the content of that block